index.js

```diff
@@ -38,33 +40,62 @@ const button = ActionButton({ // eslint-disable-line new-cap
   }
 });
 
+function setupApp() {
+  app = new Router(mod);
 
-function sendToWeb(data) {
-  mod.port.emit('from-addon-to-web', data);
-}
+  if (self.loadReason === 'install') {
+    app.send('addon-self:installed');
+  } else if (self.loadReason === 'enable') {
+    app.send('addon-self:enabled');
+  } else if (self.loadReason === 'upgrade') {
+    app.send('addon-self:upgraded');
+  }
 
-function formatInstallData(install, addon) {
-  let formatted = {
-    'name': install.name ? install.name : '',
-    'error': install.error,
-    'state': install.state,
-    'version': install.version ? install.version : '',
-    'progress': install.progress,
-    'maxProgress': install.maxProgress
-  };
+  app.on('update-check', function() {
+    app.send('addon-updates', state.updates);
+  });
 
-  if (addon) {
-    formatted = require('xtend')(formatted, {
-      'description': addon.description,
-      'homepageURL': addon.homepageURL,
-      'iconURL': addon.iconURL,
-      'size': addon.size,
-      'signedState': addon.signedState,
-      'permissions': addon.permissions
+  app.on('update-approve', function(msg) {
+    state.updates = require('exclude')(state.updates, msg);
+    button.badge = state.updates.length;
+    triggerInstalls(msg);
+  });
+
+  app.on('uninstall-all', function() {
+    uninstall(state.updates, function(addons) {
+      app.send('addons-uninstalled', addons );
     });
-  }
+    state.updates = updates.map(function(u) {
+                      return u.title;
+                    });
+  });
 
-  return formatted;
+  app.on('uninstall', function(msg) {
+    uninstall(msg, function(addons) {
+      app.send('addons-uninstalled', addons);
+    });
+    const addonCheck = msg.map(function(a) {
+                          return a.toLowerCase().trim();
+                        });
```

Comment:
If we have arrow function support, we could flatten these somewhat to something more ES6-ey like:

```js
const addonCheck = msg.map((a) => normalize(a));
updates.filter((u) => _.contains(addonCheck, normalize(u.title)))
  .map((u) => u.title)
  .forEach((u) => {
    state.updates.push(u);
  });

function normalize(str) {
  return str.toString().toLowerCase().trim();
}
```